pkg/build/api/validation/validation.go

```diff
@@ -103,11 +103,18 @@ func validateBuildSpec(spec *buildapi.BuildSpec, fldPath *field.Path) field.Erro
 	allErrs := field.ErrorList{}
 	s := spec.Strategy
 
-	if s.CustomStrategy == nil && spec.Source.Git == nil && spec.Source.Binary == nil && spec.Source.Dockerfile == nil {
+	if s.CustomStrategy == nil && s.JenkinsPipelineStrategy == nil && spec.Source.Git == nil && spec.Source.Binary == nil && spec.Source.Dockerfile == nil {
```

Comment:
@bparees good question. there are two scenarios, right?

1) I have vendored Jenkinsfile into BuildConfig. What happens when I change it? Does ConfigChange trigger redeploy the Jenkins service?

2) I have Jenkinsfile in GIT repo and I pushed update to the Jenkinsfile. Will OpenShift have Github/Git hook that will trigger the "redeployment" of Jenkins, or is this something the Jenkins plugin will manage itself? (iow. reloading of Jenkinsfile).